src/xenia/kernel/xam/content_manager.cc

```diff
@@ -138,6 +148,63 @@ bool ContentManager::ContentExists(const ContentData& data) {
   return std::filesystem::exists(path);
 }
 
+X_RESULT ContentManager::CreateContentHeaderFile(const std::string file_name,
+                                                 const XCONTENT_DATA* data) {
+  auto title_id = fmt::format("{:8X}", kernel_state_->title_id());
+  auto content_type = fmt::format("{:08X}", data->content_type);
+  auto header_path =
+      root_path_ / title_id / kGameContentHeaderDirName / content_type;
+
+  if (!std::filesystem::exists(header_path)) {
+    if (!std::filesystem::create_directories(header_path)) {
+      return X_STATUS_ACCESS_DENIED;
+    }
+  }
+  auto header_filename = file_name + ".header";
+
+  xe::filesystem::CreateFile(header_path / header_filename);
+
+  if (std::filesystem::exists(header_path / header_filename)) {
+    auto file = xe::filesystem::OpenFile(header_path / header_filename, "wb");
+    fwrite(data, 1, sizeof(XCONTENT_DATA), file);
+    fclose(file);
+    return X_STATUS_SUCCESS;
+  }
+  return X_STATUS_NO_SUCH_FILE;
+}
+
+X_RESULT ContentManager::OpenContentHeaderFile(const std::string_view file_name,
+                                               uint32_t content_type,
+                                               XCONTENT_DATA& data) {
+  auto title_id = fmt::format("{:8X}", kernel_state_->title_id());
+  auto content_type_directory = fmt::format("{:08X}", content_type);
+  auto header_file_path = root_path_ / title_id / kGameContentHeaderDirName /
+                          content_type_directory / file_name;
+  constexpr uint32_t header_size = sizeof(XCONTENT_DATA);
+
+  if (std::filesystem::exists(header_file_path)) {
+    auto file = xe::filesystem::OpenFile(header_file_path, "rb");
+
+    std::array<uint8_t, header_size> buffer;
+
+    auto fize_size = std::filesystem::file_size(header_file_path);
+    if (fize_size < header_size) {
```

Comment:
Is there an expectation that the file size could ever not be the size of `XCONTENT_DATA`?

Reply:
Tbh I should just check if it is different than ``XCONTENT_DATA`` size. I added this check in case someone will manually poke with these files.